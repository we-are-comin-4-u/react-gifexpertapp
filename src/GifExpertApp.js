import React, { useState } from 'react'
import { AddCategory } from './Components/AddCategory';
import { GifGrid } from './Components/GifGrid';
 
// const categories = ['One Punch ,', 'Samurai X, ', 'Dragon Ball']


export const GifExpertApp = ({defaultCategories = []}) => {
   // const [categories, setCategories] = useState(['One Punch ']);
   const [categories, setCategories] = useState(defaultCategories);

   // const handleAdd = () =>{
      //  setCategories (['Golden Boy',...categories]);
      //  setCategories( catego => [...catego,'Superman'] )
      
   // }
   return (
      <>
         <h2>GifExpertApp</h2>
         <AddCategory setCategories ={setCategories}/>
         <hr/>
         {/* <button onClick ={handleAdd}>âœ…</button> */}

        
            <ol>
               {
                  categories.map(category =>(
                     <GifGrid 
                     key = {category}
                     category ={category}
                     />
                     ))
               }
            </ol>
         

      </>
   )
}

export default GifExpertApp;