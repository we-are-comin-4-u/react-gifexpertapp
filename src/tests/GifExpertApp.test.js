import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';

describe('testing <GifExpertApp/> ',()=>{

   test('should to show the right snatshot',()=>{

      const wrapper = shallow( <GifExpertApp/>)

      expect(wrapper).toMatchSnapshot();
      
   })

   test('should to show a list of categories',()=>{

      const categories = ['One Punch', 'Dragon Ball']
      const wrapper = shallow( <GifExpertApp defaultCategories = {categories}/>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })



});