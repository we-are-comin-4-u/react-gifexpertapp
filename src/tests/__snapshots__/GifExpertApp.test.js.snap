// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GifExpertApp/>  should to show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`testing <GifExpertApp/>  should to show the right snatshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
